libgraph/include/katana/GraphTopology.h

```diff
@@ -1303,21 +1425,23 @@ class KATANA_EXPORT EdgeTypeAwareBiDirTopology
 
 template <typename Graph>
 using has_undirected_t =
-    decltype(std::declval<Graph>().UndirectedEdges(typename Graph::Node()));
+    decltype(std::declval<Graph>().UndirectedEdges(std::declval<typename Graph::Node>()));
 
-template <typename Graph>
-auto
-Edges(const Graph& graph, typename Graph::Node node) {
-  if constexpr (katana::is_detected_v<has_undirected_t, Graph>) {
```

Comment:
Do you prefer SFINAE to `constexpr if`? It's the same result with more code and ugly templates, unless I am missing something.

Reply:
I see you didn't remove it below, so there must be something I'm missing.

Reply:
Return types are different, hence two different overloads.

Reply:
It could be that the compiler can handle this case. I'm yet to find out.

Reply:
`auto` functions with `constexpr if` support different branches returning different types, e.g. https://godbolt.org/z/4MPcTrssK

Reply:
Disturbing :). But I don't mind if you restore to `constexpr if`.